satpy/readers/yaml_reader.py

```diff
@@ -493,6 +494,52 @@ def create_filehandlers(self, filenames):
                     filehandlers,
                     key=lambda fhd: (fhd.start_time, fhd.filename))
 
+        # update existing dataset IDs with information from the file handler
+        self.update_ds_info_from_files()
+
+        # load any additional dataset IDs determined dynamically from the file
+        self.update_ds_ids_from_files()
+
+    def update_ds_info_from_files(self):
+        """Update DatasetIDs with information from loaded files.
+
+        This is useful, for example, if dataset resolution may change
+        depending on what files were loaded.
+
+        """
```

Comment:
The name of this function is missleading as it doesn't mention the change of the id. So either it does both and should be merged with `update_ds_ids_from_files` or we should move the id part to the next function.

Reply:
The dict and the ID have to be updated at the same time or at least it makes the most sent to me that they do because the ID is made from the dict. I used to have these two methods as one single method but thought it might be better to separate them in case someone wanted to customize how the DatasetID gets updated. For example, something other than resolution needs to be updated in the DatasetID. I could also change the name of this function if it makes more sense?

Reply:
yes, maybe a name change is welcome then.

Reply:
What about the below method is `add_ds_ids_from_files` and this method is `update_ds_ids_from_files`? Or maybe `update_ds_ids_from_file_handlers`?

Reply:
`update_ds_ids_from_file_handlers` is good.